lib/Differentiator/DiffPlanner.cpp

```diff
@@ -13,6 +14,48 @@ using namespace clang;
 
 namespace clad {
   static SourceLocation noLoc;
+  
+  // Returns DeclRefExpr of function argument of differentiation calls.
+  // If argument is passed for relevantAncestor, then it's value will be 
+  // modified in-place to contain relevant ancestor of the function 
+  // argument's DeclRefExpr
+  // Here relevant ancestor is nearest ancestor of ImplicitCastExpr or 
+  // UnaryOperator type.
+  DeclRefExpr* getArgFunction(CallExpr* call,Expr** relevantAncestor = nullptr) {
+    struct Finder :
+      RecursiveASTVisitor<Finder> {
+        DeclRefExpr* m_FnDRE = nullptr;
+        Expr** m_RelevantAncestor = nullptr;
+        Finder(Expr** relevantAncestor) : m_RelevantAncestor(relevantAncestor) {}
+        bool VisitExpr(Expr* E) {
+          if (m_RelevantAncestor) {
+            switch (E->getStmtClass()) {
+              case Stmt::StmtClass::ImplicitCastExprClass:
+              case Stmt::StmtClass::UnaryOperatorClass:
+                *m_RelevantAncestor = E;
+                break; 
+            }
+          }
+          if (auto DRE = dyn_cast<DeclRefExpr>(E)) {
+            if (auto VD = dyn_cast<VarDecl>(DRE->getDecl())) {
+              TraverseStmt(VD->getInit());
+            } else if (auto FD = dyn_cast<FunctionDecl>(DRE->getDecl())) {
+              m_FnDRE = DRE;
+            }
+            return false;  
+          }
+          return true; 
+        }
+      } finder(relevantAncestor);
+
+      auto callFn = call->getDirectCallee();
+      auto cladNS = dyn_cast<NamespaceDecl>(callFn->getDeclContext());
+      assert(cladNS->getName() == "clad" &&
+        "Only clad:: function call expression can be passed as call argument");
```

Comment:
```suggestion
      assert(cast<NamespaceDecl>(call->getDirectCallee())->
             getDeclContext())->getName() == "clad" &&
             "Should be called for clad:: special functions!");
```

asserts are only expanded in debug builds and `cladNS` will become a unused variable in release.

Reply:
Oh okay, I will make this change, and I will remember this for future PRs :)